executor.go

```diff
@@ -874,8 +874,15 @@ func (r RowIDs) merge(other RowIDs, limit int) RowIDs {
 }
 
 func (e *executor) executeGroupBy(ctx context.Context, index string, c *pql.Call, shards []uint64, opt *execOptions) ([]GroupCount, error) {
+	// extract filter call.
+	var filter *pql.Call
+	children := c.Children
+	if len(children) > 0 && isBitmapCallName(children[0].Name) {
+		filter, children = children[0], children[1:]
+	}
```

Comment:
This splits off the `filter` from the rest of the `Rows()` children. These are propagated through the shard calls later.